metricflow/sql/render/trino.py

```diff
@@ -0,0 +1,124 @@
+        return {
+            SqlPercentileFunctionType.APPROXIMATE_CONTINUOUS,
+        }
+
+    @override
+    def visit_time_delta_expr(self, node: SqlTimeDeltaExpression) -> SqlExpressionRenderResult:
+        """Render time delta expression for Trino, which requires slightly different syntax from other engines."""
+        arg_rendered = node.arg.accept(self)
+        if node.grain_to_date:
+            return SqlExpressionRenderResult(
+                sql=f"DATE_TRUNC('{node.granularity.value}', TIMESTAMP {arg_rendered.sql})",
+                bind_parameters=arg_rendered.bind_parameters,
+            )
+
+        count = node.count
+        granularity = node.granularity
+        if granularity == TimeGranularity.QUARTER:
+            granularity = TimeGranularity.MONTH
+            count *= 3
+
+        # Trino interval needs to be in quotes.
+        return SqlExpressionRenderResult(
+            sql=f"CAST({arg_rendered.sql} AS {self.timestamp_data_type}) - INTERVAL '{count}' {granularity.value}",
+            bind_parameters=arg_rendered.bind_parameters,
+        )
+
+    @override
+    def visit_generate_uuid_expr(self, node: SqlGenerateUuidExpression) -> SqlExpressionRenderResult:
+        return SqlExpressionRenderResult(
+            sql="uuid()",
+            bind_parameters=SqlBindParameters(),
+        )
+
+    @override
+    def visit_percentile_expr(self, node: SqlPercentileExpression) -> SqlExpressionRenderResult:
+        """Render a percentile expression for Trino."""
+        arg_rendered = self.render_sql_expr(node.order_by_arg)
+        params = arg_rendered.bind_parameters
+        percentile = node.percentile_args.percentile
+
+        if node.percentile_args.function_type is SqlPercentileFunctionType.CONTINUOUS:
+            function_str = "PERCENTILE_CONT"
+        elif node.percentile_args.function_type is SqlPercentileFunctionType.DISCRETE:
+            function_str = "PERCENTILE_DISC"
+        elif node.percentile_args.function_type is SqlPercentileFunctionType.APPROXIMATE_CONTINUOUS:
+            return SqlExpressionRenderResult(
+                sql=f"approx_percentile({arg_rendered.sql}, {percentile})",
+                bind_parameters=params,
+            )
+        elif node.percentile_args.function_type is SqlPercentileFunctionType.APPROXIMATE_DISCRETE:
+            raise RuntimeError(
+                "Approximate discrete percentile aggregatew not supported for Trino. Set "
+                + "use_discrete_percentile and/or use_approximate_percentile to false in all percentile measures."
+            )
+        else:
+            assert_values_exhausted(node.percentile_args.function_type)
+
+        return SqlExpressionRenderResult(
+            sql=f"{function_str}({percentile}) WITHIN GROUP (ORDER BY ({arg_rendered.sql}))",
+            bind_parameters=params,
+        )
+
+    @override
+    def visit_between_expr(self, node: SqlBetweenExpression) -> SqlExpressionRenderResult:  # noqa: D
```

Comment:
Can we add a docstring to indicate why Trino requires this override? Seems like the date/time literals need to be cast to timestamp. Is that all?